### PR TITLE
e2e: Replace Kuttl-in-Kuttl

### LINKED ISSUE #103: e2e: Replace Kuttl-in-Kuttl with simple kubectl apply in scripts
The Kuttl-in-Kuttl pattern introduced in https://github.com/linode/cluster-api-provider-linode/pull/80 can be replaced with defining a script that simply runs `kubectl apply` while passing dynamic data. The `TestStep` will fail if any of the commands exit with non-zero.

```sh
MACHINE_UID="$(OBJ=machines/machine-sample make getKubeUid)"

cat <<EOF | kubectl apply -n $NAMESPACE -f -
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha1
kind: LinodeMachine
metadata:
  ownerReferences:
  - apiVersion: cluster.x-k8s.io/v1beta1
    kind: Machine
    name: machine-sample
    uid: ${MACHINE_UID}
EOF
```

### PR DESCRIPTION
**Pull request by bcmendoza820**
Closes #103

This PR replaces kuttl-in-kuttl and removes related helper scripts.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -324,12 +322,6 @@ envtest: $(ENVTEST) ## Download setup-envtest locally if necessary.
 $(ENVTEST): $(LOCALBIN)
 	GOBIN=$(CACHE_BIN) go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
 
-.PHONY: envsubst
```
Comment (AshleyDumaine) [inline Makefile:327]: We should keep envsubst, we're using that in the Tiltfile: https://github.com/linode/cluster-api-provider-linode/blob/main/Tiltfile#L16

**Review by AshleyDumaine — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (eljohnson92) [reply Makefile]: @AshleyDumaine  if we only need envsubst for the Tiltfile we should absolutely rip it out fo there as well. is there anywhere else we are using envsubst explicitly?

**Review by AshleyDumaine — APPROVED** [review]

**Comment by eddycharly** [comment]
Folks did you look at my PR #119 ?

We can discuss the pros/cons if you want, there's a lot of things to win switching to chainsaw IMHO.

**Comment by eljohnson92** [comment]
Hi @eddycharly we wanted to first remove the kuttl-in-kuttl pattern before considering switching tools so we can compare with a simpler use-case.

**Comment by eddycharly** [comment]
@eljohnson92 👍

**Pull request merged**